torch/lib/THC/THCTensorIndex.cu

```diff
@@ -373,6 +375,7 @@ __device__ __forceinline__ long calculateOffset(
       indexAtDim = index - nextIndex * sizeAtDim;
     }
 
+    assert(indexAtDim < data.baseSizes[dim]);
```

Comment:
In general we don't do >= 0 tests in CuTorch, see e.g: https://github.com/torch/cutorch/blob/master/lib/THC/THCTensorIndex.cu#L38 and others in that file, because we have `typename IndexType` as an `unsigned int` or `unsigned long`.